File: hydrograph.engine/hydrograph.engine.cascading/src/main/java/hydrograph/engine/adapters/InputOracleAdapter.java
package hydrograph.engine.adapters;

import hydrograph.engine.adapters.base.InputAdapterBase;
import hydrograph.engine.cascading.assembly.InputOracleAssembly;
import hydrograph.engine.cascading.assembly.base.BaseComponent;
import hydrograph.engine.cascading.assembly.infra.ComponentParameters;
import hydrograph.engine.core.component.generator.InputOracleEntityGenerator;
import hydrograph.engine.jaxb.commontypes.TypeBaseComponent;

public class InputOracleAdapter extends InputAdapterBase {

	InputOracleAssembly oracleComponent;
	InputOracleEntityGenerator entityGenerator;

	public InputOracleAdapter(TypeBaseComponent baseComponent) {
		entityGenerator = new InputOracleEntityGenerator(baseComponent);
	}

	@Override
	public BaseComponent getAssembly() {
		return oracleComponent;
	}

	@Override
	public void createAssembly(ComponentParameters componentParameters) {
		oracleComponent = new InputOracleAssembly(entityGenerator.getEntity(), componentParameters);

	}

}

File: hydrograph.engine/hydrograph.engine.spark/src/main/scala/hydrograph/engine/spark/components/adapter/InputOracleAdapter.scala
package hydrograph.engine.spark.components.adapter

import hydrograph.engine.core.component.generator.InputOracleEntityGenerator
import hydrograph.engine.jaxb.commontypes.TypeBaseComponent
import hydrograph.engine.spark.components.OracleInputComponent
import hydrograph.engine.spark.components.adapter.base.InputAdatperBase
import hydrograph.engine.spark.components.base.InputComponentBase
import hydrograph.engine.spark.components.platform.BaseComponentParams


class InputOracleAdapter(typeBaseComponent: TypeBaseComponent) extends InputAdatperBase {

  var inputOracleEntityGenerator: InputOracleEntityGenerator = null
  var inputOracleComponent: OracleInputComponent = null;

  override def getComponent(): InputComponentBase = inputOracleComponent

  override def createGenerator(): Unit = {
    inputOracleEntityGenerator = new InputOracleEntityGenerator(typeBaseComponent)
  }

  override def createComponent(baseComponentParams: BaseComponentParams): Unit = {
    inputOracleComponent = new OracleInputComponent(inputOracleEntityGenerator.getEntity, baseComponentParams)
  }
}
